### ISSUE TITLE
Error: ENOTDIR: not a directory, scandir (tslint 5.3.0)

### ISSUE THREAD

**Issue by wub** [palantir/tslint#2811]
### Bug Report

- __TSLint version__: 5.3.0
- __TypeScript version__: 2.3.3
- __Running TSLint via__: webpack

#### TypeScript code being linted

```ts
// Any typescript file, was working with 5.2.0
```

with `tslint.json` configuration:

```json
{
    "extends": [
        "tslint-eslint-rules"
    ],
    "rules": {
        "adjacent-overload-signatures": true,
        "align": [true, "parameters"],
        "no-var-keyword": true,
        "no-bitwise": true,
        "no-shadowed-variable": true,
        "no-inferrable-types": [true, "ignore-params", "ignore-properties"],
        "no-arg": true,
        "forin": true,
        "curly": true,
        "eofline": true,
        "no-eval": true,
        "no-empty": true,
        "no-construct": true,
        "no-invalid-this": true,
        "no-duplicate-super": true,
        "no-duplicate-variable": true,
        "no-consecutive-blank-lines": [true],
        "no-conditional-assignment": true,
        "object-literal-shorthand": true,
        "one-variable-per-declaration": [true],
        "no-unused-expression": true,
        "class-name": true,
        "max-file-line-count": [true, 800],
        "max-line-length": [true, 200],
        "quotemark": [true, "double"],
        "prefer-for-of": true,
        "newline-before-return": true
    }
}

```

#### Actual behavior
```
ERROR in ./Scripts/pages/profileDetail/ProfileDetail.tsx
Module build failed: Error: ENOTDIR: not a directory, scandir 'D:\Code\jm-featured-photos\StarNow.Web\Scripts\pages\profileDetail\ProfileDetail.tsx'
    at Object.fs.readdirSync (fs.js:914:18)
    at findFile (D:\Code\jm-featured-photos\StarNow.Web\node_modules\tslint\lib\configuration.js:120:25)
    at findup (D:\Code\jm-featured-photos\StarNow.Web\node_modules\tslint\lib\configuration.js:103:19)
    at findConfigurationPath (D:\Code\jm-featured-photos\StarNow.Web\node_modules\tslint\lib\configuration.js:80:30)
    at Function.findConfiguration (D:\Code\jm-featured-photos\StarNow.Web\node_modules\tslint\lib\configuration.js:48:16)
    at parseConfigFile (D:\Code\jm-featured-photos\StarNow.Web\node_modules\tslint-loader\index.js:44:24)
    at resolveOptions (D:\Code\jm-featured-photos\StarNow.Web\node_modules\tslint-loader\index.js:35:27)
    at Object.module.exports (D:\Code\jm-featured-photos\StarNow.Web\node_modules\tslint-loader\index.js:139:17)
 @ multi ./Scripts/pages/profileDetail/ProfileDetail.tsx
```
This error appears once for every ts/tsx file.

#### Expected behavior
Successful compilation.

**Comment by pe8ter** [comment]
For an extra data point on this bug, gulp-tslint@8.0.0 has the same issue (`ENOTDIR`). You can see that gulp-tslint [passes the path to the file being linted](https://github.com/panuhorsmalahti/gulp-tslint/blob/5a16f03782a831d7c8f6bd6ceccd800f3b5b25cd/index.js#L101), as opposed to the path to the directory that contains the file. Doing so worked in tslint@5.2.0, but now fails in tslint@5.3.0.

I verified that by manually changing `file.path` to `file.base` in gulp-tslint that linting works again.

Perhaps I should file a bug with gulp-tslint? Or is this the sort of breaking change not desired in a minor version release of tslint?

**Comment by nchen63** [comment]
Shouldn't be breaking on minor versions. Will take a look

**Comment by adidahiya** [comment]
I'm experiencing this with vscode-tslint too :(

**Comment by gucong3000** [comment]
Same thing happend to me.
https://travis-ci.org/gucong3000/gulp-reporter/builds/235020961

- __TSLint version__: ^5.3.0
- __TypeScript version__: ^2.2.2
- __Running TSLint via__: gulp-tslint ^8.0.0
